src/helpers/functions.ts

```diff
@@ -58,3 +61,54 @@ export const waterfall = <TFns extends ((arg?: any) => any)[]>(
     return chain;
   };
 };
+
+type ParseErr = string;
+export const parseFnData = async (
+  data: unknown,
+  api: InngestAPI
+): Promise<Result<FnData, ParseErr>> => {
+  try {
+    const result = fnDataSchema.parse(data);
+
+    if (result.use_api) {
+      const [evtResp, stepResp] = await Promise.all([
+        api.getRunBatch(result.ctx?.run_id as string),
+        api.getRunSteps(result.ctx?.run_id as string),
+      ]);
+
+      if (evtResp.ok) {
+        result.events = evtResp.value;
+      } else {
+        return Err(
+          prettyError({
+            whatHappened: "failed to retrieve list of events",
+            consequences: "function execution can't continue",
+            stack: true,
```

Comment:
For the `prettyError()` calls in this function, it'd be great to be able to give users some more information about how to proceed. In this case, function execution can't continue, but what's my course of action? Was it my fault? Have my events been lost?

There are a few extra properties we can use to further describe the problem and how to fix it:

```ts
/**
 * If applicable, provide a full sentence to reassure the user about certain
 * details, for example if an error occurred whilst uploading a file, but we
 * can assure the user that uploading succeeded and something internal failed.
 */
reassurance?: string;

/**
 * Tell the user why the error happened if we can. This should be a full
 * sentence or paragraph that explains the error in more detail, for example
 * to explain that a file failed to upload because it was too large and that
 * the maximum size is 10MB.
 */
why?: string;

/**
 * If we can, tell the user what they can do to fix the error now. This should
 * be a full sentence or paragraph that explains what the user can do to fix
 * the error, for example to tell them to try uploading a smaller file or
 * upgrade to a paid plan.
 */
toFixNow?: string | string[];

/**
 * If applicable, tell the user what to do if the error persists, they want
 * more information, or the fix we've given them doesn't work.
 *
 * This should be a full sentence or paragraph, and will likely refer users
 * to contact us for support, join our Discord, or read documentation.
 */
otherwise?: string;

/**
 * If applicable, provide a code that the user can use to reference the error
 * when contacting support.
 */
code?: ErrCode;
```

A new `code` for each unique error would be great for allowing us to later provide docs pages and shortlinks for each of these.

It may be that we can't specify all of these properties or be all that helpful for a user right at this second; filling out these fields is pretty difficult, thinking about what the user should actually do in this situation. The _least_ we can do is provide reassurance and ask them to reach out on Discord.

Reply:
there's really nothing the user can do since none of these errors will be within a user's control. it's best for us to be notified automatically somehow when these errors happened so we can look into it on our end.